### PR TITLE
feat: add collection resource list page and initial create collection page

### PR DESCRIPTION
**Pull request by karrui**
This PR introduces several components and functionalities for managing collections in the dashboard. It includes the implementation of the `CollectionBanner`, `CollectionTable`, `CreateCollectionPageModal`, and various related components. Additionally, there are updates to the `Datatable` component to enhance functionality, and new utility hooks and schemas to support the creation and management of collection pages.

---

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/438?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#438** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/438?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#426** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/426?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by seaerchin** [comment]
will review after rebase!

File: apps/studio/src/schemas/page.ts [hunk 1]
```diff
@@ -67,3 +67,32 @@ export const createPageSchema = z.object({
   // NOTE: implies that top level pages are allowed
   folderId: z.number().min(1).optional(),
 })
+
+export const createCollectionPageFormSchema = z
+  .discriminatedUnion("type", [
+    z.object({
+      type: z.literal("pdf"),
+      url: z.string().url(),
```
Comment (karrui) [inline apps/studio/src/schemas/page.ts:75]: maybe call this pdfUrl

File: apps/studio/src/server/modules/collection/collection.router.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }),
+  createCollectionPage: protectedProcedure
+    .input(createCollectionPageSchema)
+    .mutation(async ({ input }) => {
+      let newPage: PrismaJson.BlobJsonContent
+      const { title, type, permalink, siteId, collectionId } = input
+      if (type === "page") {
+        newPage = createCollectionPageJson({ title, type })
+      } else {
+        newPage = createCollectionPdfJson({ title, type, url: input.url })
```
Comment (karrui) [inline apps/studio/src/server/modules/collection/collection.router.ts:41]: todo: update json generation

**Review by seaerchin — APPROVED** [review]

**Pull request merged**